src/reconcile/rec_row.c

```diff
@@ -566,6 +566,10 @@ __rec_row_leaf_insert(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins)
 
     upd = NULL;
 
+    /* About to do some visibility checks. Update our snapshot for eviction. */
+    if (F_ISSET(session, WT_SESSION_INTERNAL) && F_ISSET(session->txn, WT_TXN_RUNNING) &&
+      F_ISSET(r, WT_REC_EVICT))
+        __wt_txn_bump_snapshot(session);
```

Comment:
Why this API `__wt_txn_bump_snapshot` and why not `__wt_txn_get_snapshot` is used? Any comment explaining the choice will be helpful.

Reply:
We don't want to pin anything for eviction threads transactions. Using  `__wt_txn_bump_snapshot()` helps with not publishing the shared transaction information.
I'll update the comments.

Reply:
Why did you choose to bump the snapshot at all during reconciliation? Were there cases when it started lagging meaningfully during a reconciliation?

Reply:
I don't have data to support bumping the transaction but it can be useful for workloads which have a number of transactions committing in parallel. Note that `__wt_txn_bump_snapshot()` is intelligent enough to avoid any work if there is no change in state of transactions since the snapshot was acquired or last updated.